src/main/java/com/netflix/dgs/plugin/services/internal/GraphQLSchemaRegistry.java

```diff
@@ -39,18 +39,22 @@ public GraphQLSchemaRegistry(Project project) {
     Optional<PsiElement> psiForSchemaType(@NotNull PsiElement psiElement, @Nullable String parentType, @Nullable String field) {
 
         TypeDefinitionRegistry registry = getRegistry(psiElement);
-        ObjectTypeDefinition type = getTypeDefinition(registry, parentType);
-        if (type != null) {
-            Optional<FieldDefinition> schemaField = type.getFieldDefinitions().stream().filter(f -> f.getName().equals(field)).findAny();
+        ObjectTypeDefinition objectType = getTypeDefinition(registry, parentType);
+        InterfaceTypeDefinition interfaceType = getInterfaceTypeDefinition(registry, parentType);
```

Comment:
probalby shouldn't pre-emptively look for the interface type, but only do that if no object type was found?

Reply:
yeah, I can change that. Was trying to make the if else branching a little less nested.